Classes/Domain/Repository/GlossaryRepository.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace WebVision\WvDeepltranslate\Domain\Repository;
+
+use DateTimeImmutable;
+use TYPO3\CMS\Core\Database\Connection;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+
+class GlossaryRepository
+{
+    /**
+     * @param int $uid
+     * @return array{name: string, id: string, source_lang: string, target_lang: string, entries: array}
+     */
+    public function getGlossaryInformationForSync(int $uid): array
+    {
+        $glossaryInformation = [
+            'entries' => [],
+        ];
+        $db = GeneralUtility::makeInstance(ConnectionPool::class)
```

Comment:
injecting ConnectionPool or creating a property seems useful to me, as it is used multiple timees

Reply:
as long as v9 is involved, DI is not really possible - needs fallback to GU::mI() then .. but generally a valid point.